**<u>Introduction</u>**

This project analyzes the S&P 500 companies using SQL for data processing and Power BI for visualization. The goal is to uncover market trends, sector distributions, and financial insights.

In [1]:
use chidiszn; -- the database name.

SELECT TOP (10) [Exchange]
      ,[Symbol]
      ,[Shortname]
      ,[Longname]
      ,[Sector]
      ,[Industry]
      ,[Currentprice]
      ,[Marketcap]
      ,[Ebitda]
      ,[Revenuegrowth]
      ,[City]
      ,[State]
      ,[Country]
      ,[Fulltimeemployees]
      ,[Longbusinesssummary]
      ,[Weight] AS sp500_companies_cleaned
  FROM [chidiszn].[dbo].[sp500_companies] -- Lets look at the data.



(10 rows affected)

Total execution time: 00:00:00.034

Exchange Symbol Shortname Longname Sector Industry Currentprice Marketcap Ebitda Revenuegrowth City State Country Fulltimeemployees Longbusinesssummary sp500_companies_cleaned NMS AAPL Apple Inc. Apple Inc. Technology Consumer Electronics 254.49 3846819807232 134660997120 0.061 Cupertino CA United States 164000 Apple Inc. designs, manufactures, and markets smartphones, personal computers, tablets, wearables, and accessories worldwide. The company offers iPhone, a line of smartphones; Mac, a line of personal computers; iPad, a line of multi-purpose tablets; and wearables, home, and accessories comprising AirPods, Apple TV, Apple Watch, Beats products, and HomePod. It also provides AppleCare support and cloud services; and operates various platforms, including the App Store that allow customers to discover and download applications and digital content, such as books, music, video, games, and podcasts, as well as advertising services include third-party licensing arrangements and its own advertising platforms. In addition, the company offers various subscription-based services, such as Apple Arcade, a game subscription service; Apple Fitness+, a personalized fitness service; Apple Music, which offers users a curated listening experience with on-demand radio stations; Apple News+, a subscription news and magazine service; Apple TV+, which offers exclusive original content; Apple Card, a co-branded credit card; and Apple Pay, a cashless payment service, as well as licenses its intellectual property. The company serves consumers, and small and mid-sized businesses; and the education, enterprise, and government markets. It distributes third-party applications for its products through the App Store. The company also sells its products through its retail and online stores, and direct sales force; and third-party cellular network carriers, wholesalers, retailers, and resellers. Apple Inc. was founded in 1976 and is headquartered in Cupertino, California. 0.06920915243972749 NMS NVDA NVIDIA Corporation NVIDIA Corporation Technology Semiconductors 134.7 3298803056640 61184000000 1.224 Santa Clara CA United States 29600 NVIDIA Corporation provides graphics and compute and networking solutions in the United States, Taiwan, China, Hong Kong, and internationally. The Graphics segment offers GeForce GPUs for gaming and PCs, the GeForce NOW game streaming service and related infrastructure, and solutions for gaming platforms; Quadro/NVIDIA RTX GPUs for enterprise workstation graphics; virtual GPU or vGPU software for cloud-based visual and virtual computing; automotive platforms for infotainment systems; and Omniverse software for building and operating metaverse and 3D internet applications. The Compute & Networking segment comprises Data Center computing platforms and end-to-end networking platforms, including Quantum for InfiniBand and Spectrum for Ethernet; NVIDIA DRIVE automated-driving platform and automotive development agreements; Jetson robotics and other embedded platforms; NVIDIA AI Enterprise and other software; and DGX Cloud software and services. The company's products are used in gaming, professional visualization, data center, and automotive markets. It sells its products to original equipment manufacturers, original device manufacturers, system integrators and distributors, independent software vendors, cloud service providers, consumer internet companies, add-in board manufacturers, distributors, automotive manufacturers and tier-1 automotive suppliers, and other ecosystem participants. NVIDIA Corporation was incorporated in 1993 and is headquartered in Santa Clara, California. 0.059349638157321576 NMS MSFT Microsoft Corporation Microsoft Corporation Technology Software - Infrastructure 436.6 3246068596736 136551997440 0.16 Redmond WA United States 228000 Microsoft Corporation develops and supports software, services, devices and solutions worldwide. The Productivity and Business Processes segment offers office, exchange, Shar

Commands completed successfully.

Total execution time: 00:00:00

### **<u>Data Cleaning Steps</u>**

- Handled missing values
    
- Ensured correct data types
    
- Standardized column names (optional for SQL use)
    

Then I wrote an SQL script assuming this data is in a table named `sp500_companies`.

  

**<u>Data Processing & Queries</u>**

In [3]:
SELECT COUNT(*) FROM sp500_companies; -- Total companies

(1 row affected)

Total execution time: 00:00:00.010

(No column name) 502

The total number of S&P 500 companies were 502 on the dataset collected for analysis.

In [4]:
SELECT DISTINCT sector FROM sp500_companies; -- List of sectors

(11 rows affected)

Total execution time: 00:00:00.167

sector Basic Materials Communication Services Consumer Cyclical Consumer Defensive Energy Financial Services Healthcare Industrials Real Estate Technology Utilities

The code above shows the sectors each companies falls under.

In [5]:
SELECT sector, AVG(revenuegrowth) AS avg_growth
FROM sp500_companies
GROUP BY sector
ORDER BY avg_growth DESC;

Warning: Null value is eliminated by an aggregate or other SET operation.

(11 rows affected)

Total execution time: 00:00:00.084

sector avg_growth Financial Services 0.12295522388059701 Technology 0.11582716049382716 Real Estate 0.08126666666666665 Healthcare 0.07709677419354836 Consumer Cyclical 0.06743636363636363 Communication Services 0.05247619047619047 Industrials 0.045142857142857144 Utilities 0.044156249999999994 Basic Materials 0.0194090909090909 Consumer Defensive 0.010297297297297298 Energy 0.007727272727272729

From Basic analysis, we can see that Financial Services lies as the biggest sector using Average groth rate.

In [6]:
SELECT SecSta = state, COUNT(*) AS company_count FROM sp500_companies GROUP BY state ORDER BY company_count DESC;

(42 rows affected)

Total execution time: 00:00:00.041

SecSta company_count CA 67 NY 49 TX 48 IL 32 OH 22 MA 22 PA 20 NULL 20 FL 17 GA 17 VA 17 MN 15 NJ 15 CT 14 NC 13 WA 12 AZ 10 MI 9 TN 8 CO 8 IN 8 MO 7 MD 7 WI 7 NV 4 RI 4 OK 4 KY 3 AR 3 DC 3 DE 2 AL 2 ID 2 LA 2 NE 2 NH 1 ME 1 MT 1 OR 1 IA 1 BC 1 UT 1

**Company Distribution by State**

In [7]:
SELECT sector, COUNT(*) AS company_count
FROM sp500_companies
GROUP BY sector
ORDER BY company_count DESC;

(11 rows affected)

Total execution time: 00:00:00.018

sector company_count Technology 82 Industrials 70 Financial Services 67 Healthcare 62 Consumer Cyclical 55 Consumer Defensive 37 Utilities 32 Real Estate 31 Energy 22 Basic Materials 22 Communication Services 22

**<u>Sector distribution</u>**

We've done data cleaning and descriptive analysis. Now we create a python scriptfor the data analysis.